File: turbo/src/main/java/com/androidpi/turbo/utils/AppProfileHelper.java
package com.androidpi.turbo.utils;

import android.content.Context;
import android.graphics.drawable.Drawable;

/**
 * Created by yinhang on 2018/12/5
 */
public class AppProfileHelper {

    public static Drawable getIconDrawable(Context context) {
        int iconRes = context.getApplicationInfo().icon;
        if (iconRes == 0) {
            return context.getApplicationInfo().loadIcon(context.getPackageManager());
        }
        return context.getResources().getDrawable(iconRes);
    }
}

File: turbo/src/main/java/com/androidpi/turbo/utils/AppProfileHelper.kt
package com.androidpi.turbo.utils

import android.content.Context
import android.graphics.drawable.Drawable

/**
 * Created on 2018/12/5
 */
object AppProfileHelper {
    fun getIconDrawable(context: Context): Drawable {
        val iconRes = context.applicationInfo.icon
        return if (iconRes == 0) {
            context.applicationInfo.loadIcon(context.packageManager)
        } else context.resources.getDrawable(iconRes)
    }
}